### PR TITLE
Add Swagger Docs on Github pages

### LINKED ISSUE #14: Better documentation
I think we need to improve the documentation of this project. Having all the important stuff and the details together in the README is a bit messy and not clear to new users. Also, we should add more information and guides on how to build and deploy the package and be clear about the new features added to upstream.

To address this, we could use the integrated GitHub Wiki and link the pages in README but if you have another proposal please share it in comments. :+1:

### PR DESCRIPTION
**Pull request by code1mountain**
Closes #14.

With this pull request we can create a swagger documentation on github pages. Should be applied together with #25 , because all the methods from this pull request are documented here.

It also has a github action to update swagger UI from time to time and send a pull request automatically. I had to modify it, but I think it works (had no chance to try it out).

The documentation can be viewed at [https://code1mountain.github.io/tdlight-telegram-bot-api/](https://code1mountain.github.io/tdlight-telegram-bot-api/). It will later be available at [https://tdlight-team.github.io/tdlight-telegram-bot-api/](https://tdlight-team.github.io/tdlight-telegram-bot-api/). 

Activating github pages for the first time takes a few minutes, after that every change should be ready after about a minute (because of the github cdn).

### CODE REVIEW

**Comment by MarcoBuster** [comment]
Do you have an example of the auto-PR working?

**Comment by code1mountain** [comment]
I'm trying to trigger one manually, one moment...

**Comment by code1mountain** [comment]
image
I managed to get it to work by setting the version back manually. It sends a Pull request like [this one](https://github.com/code1mountain/tdlight-api-docs/pull/1)

**Comment by MarcoBuster** [comment]
@code1mountain status? Waiting for review?

**Comment by code1mountain** [comment]
Yes, and also waiting for the other PR to be applied at the same time

**Comment by code1mountain** [comment]
Just saw that it got merged, we can merge this PR now I think. Is there still confusion if we want the documentation files in this repo or a separate one?

**Comment by MarcoBuster** [comment]
> Just saw that it got merged, we can merge this PR now I think. Is there still confusion if we want the documentation files in this repo or a separate one?

@Cavallium what do you think?

File: .github/workflows/validate-swagger.yml
Comment (MarcoBuster) [inline .github/workflows/validate-swagger.yml:1]: I would change this to "Documentation" and rename the file to "docs.yml". The name of a step should be as broad as possible, the information on what is currently doing is contained in the job.

**Review by MarcoBuster — CHANGES_REQUESTED** [review]

**Review by Cavallium — CHANGES_REQUESTED** [review]
Remove "dist" directory from this repository, also add it in .gitignore.
Generated files shouldn't be on the repository.

**Comment by Cavallium** [comment]
@MarcoBuster That we should not put "dist" directory in this repository.
I can host them if you want.

**Comment by Cavallium** [comment]
@MarcoBuster I think that it's ok to put documentation in this repository, but only the source of the documentation.
Generated files and generated webpages shouldn't be on the repository.
If you want I can host the documentation, or we can use github pages. But not here on the master branch in a subfolder...

**Comment by MarcoBuster** [comment]
@Cavallium Sure, that's fair.

**Comment by code1mountain** [comment]
We could also use GitHub actions and have all the generated/downloaded stuff in the gh-pages branch and host it with `peaceiris/actions-gh-pages@v3`
Then I would not check for an updated version on a schedule but just use the latest version on every commit. An other option is to have a separate repo with the generated/downloaded stuff and only the yaml file is in the repo.

The downside with this is that everyone who wants to edit the documentation and verify that it is still valid needs some external software, but https://editor.swagger.io also does that.

**Comment by code1mountain** [comment]
The nice thing with the GitHub pages action and the hosting in a separate branch is that every fork would have its own hosted documentation with all the changes made to it (if GitHub pages is enabled on the fork ofc).

**Comment by code1mountain** [comment]
Please review @Cavallium @MarcoBuster 
All the html stuff is now in a separate branch called `gh-pages` that gets created with the latest version of swagger UI on every push to the main branch of this repository (not pull requests or anything else). Only the `tdlight-api-openapi.yaml` file and the `icons/` folder are left in the main branch. The documentation is checked for validity on every push and pull request.

**Comment by Cavallium** [comment]
@code1mountain The branch is not "main" but it's "master"

**Review by Cavallium — CHANGES_REQUESTED** [review]
Fix branch name

**Comment by code1mountain** [comment]
Okay, now it's working on the master branch

**Review by Cavallium — APPROVED** [review]
It's ok for me, I'll wait @MarcoBuster to merge

**Comment by Cavallium** [comment]
@luckydonald Ok

**Comment by luckydonald** [comment]
Additionally, how would we handle pull requests with only a single branch? Would it automatically create a new `gh-pages_brach123` for those?
 
How would the workflow for those branches look like?

Because my thinking is the following: 
If you have to run the tool locally _anyway_ to make sure everything is generating correctly before a merge of a PR (which should be checked, no?),
checking in those files in the same branch wouldn't really be any additional work.

**Comment by code1mountain** [comment]
> What I dislike about a separate branch is finding the correct docs for am older version would be incredible difficult.
> 
> Maybe we can make the automated commits at least contain the original commit hash so one has a chance to find it later?

The separate branch mainly consists of a bunch of static html files and only the yaml file gets copied over there. If you want an older version of the docs, you can use the yaml file that sits in the master branch and view it with any swagger UI you want like https://editor.swagger.io. If we adapt to proper versioning in the future we will find a way to host the documentation for every release, but for now only the latest documentation is hosted. You can also copy the link to the raw openapi yaml file from a past version and enter it into the top bar of the documentation to view it.

**Comment by code1mountain** [comment]
> Additionally, how would we handle pull requests with only a single branch? Would it automatically create a new `gh-pages_brach123` for those?
> 
> How would the workflow for those branches look like?
> 
> Because my thinking is the following:
> If you have to run the tool locally _anyway_ to make sure everything is generating correctly before a merge of a PR (which should be checked, no?),
> checking in those files in the same branch wouldn't really be any additional work.

The workflow for branches overwrites the same branch `gh-pages` for every push on master that changes the openapi yaml file. The documentation in new pull requests can be displayed like described in the last message, and it get's checked for syntax errors automatically.

**Review by Cavallium — APPROVED** [review]

**Pull request merged**
